### PR TITLE
Add Azure boot up test and clean up Azure resources after test

### PR DESCRIPTION
**Pull request by yuxisun1217**
1. Boot up Azure VM and check image
2. Delete os_disk, NIC, public-ip, nsg, vnet, image and blob in cleanupAzure.

Signed-off-by: Yuxin Sun <yuxisun@redhat.com>

### CODE REVIEW

**Comment by yuxisun1217** [comment]
The Azure test has passed. AWS stage failed because of #199 .

File: test/cases/api.sh [hunk 1]
```diff
@@ -52,12 +52,28 @@ function cleanupAzure() {
...
+    NIC_ID=$(jq -r '.networkProfile.networkInterfaces[0].id' "$WORKDIR"/vm_details.json)
+    "$AZURE_CMD" network nic show --ids "$NIC_ID" > "$WORKDIR"/nic_details.json
+    NSG_ID=$(jq -r '.networkSecurityGroup.id' "$WORKDIR"/nic_details.json)
+    PUBLICIP_ID=$(jq -r '.ipConfigurations[0].publicIpAddress.id' "$WORKDIR"/nic_details.json)
+    VNET_ID=$(jq -r '.ipConfigurations[0].subnet.id' "$WORKDIR"/nic_details.json | cut -d '/' -f1-9)
+
+    # Delete resources one by one
+    for id in "$VM_ID" "$OSDISK_ID" "$NIC_ID" "$NSG_ID" "$PUBLICIP_ID" "$VNET_ID"; do
+      echo "Deleting $id..."
+      "$AZURE_CMD" resource delete --ids "$id"
```
Comment (atodorov) [inline test/cases/api.sh:70]: Does the order in which resources are deleted matter?  If yes its a good idea to document that with a comment.

File: test/cases/api.sh [hunk 2]
```diff
@@ -533,6 +549,31 @@ function Test_verifyComposeResultAzure() {
   # https://github.com/Azure/azure-cli/issues/17123
   # Without this issue fixed or worked around, I'm not able to delete the disk
   # attached to the VM.
+
```
Comment (atodorov) [inline test/cases/api.sh:552]: Remove commented section above.

**Review by atodorov — CHANGES_REQUESTED** [review]

File: test/cases/api.sh (hunk 1 above)
Comment (yuxisun1217) [reply test/cases/api.sh:72]: Yes. I'll add comment. Besides, I just aware that the vnet will be reused by other VMs in the same resource group so I change to not delete it. Due to MSFT's document "Virtual Network in Azure is free of charge.", so it won't be charged if no traffic.
Just updated. Please help to review again. Thanks!

**Review by atodorov — APPROVED** [review]

**Review by atodorov — APPROVED** [review]

**Pull request merged**